### ISSUE TITLE
End Gateways in End dimensions created with multiverse don't teleport players anywhere [1.16.1]

### ISSUE THREAD

**Issue by jinkhya** [Multiverse/Multiverse-Core#2331]
Hello,
I created an end dimension with multiverse, slayed the dragon and when I use an enderpearl in the gateway nothing happens. This issue does not happen in the original end dimension.

Paper: git-Paper-44
Multiverse plugins :
- Multiverse-Core-4.1.1-SNAPSHOT.jar 
- Multiverse-Inventories-4.0.0-SNAPSHOT.jar
- Multiverse-NetherPortals-4.1.1-SNAPSHOT.jar

**Comment by kakd** [comment]
Same for me on Purpur 1.16.1

- In the world_the_end (generated on 1.15.2 with worldborder), everything works fine
- In another end world called resend (generated on 1.16.1, not fully), players who try to go through an end gateway to get out of the main island don't go to the outer islands. One gateway teleports them right below the end gateway where they fall to their deaths and the other doesn't teleport them at all.
- If they try to return to the main island by entering the corresponding portal on the outer islands, they are teleported back as expected.

**Comment by SharasGit** [comment]
I confirm the problem. I created an END world with Multiverse 4.1.0 on 1.16.1 with paper-79. After killing the dragon, when i use an enderpearl in the gateway, sometimes nothing happens, sometimes i get teleported near the gateway, or inside, or on the main island, but i never leave the main island.

**Comment by nicegamer7** [comment]
Please give this a try with the latest Spigot from BuildTools or Paper <span>#</span>83.

**Comment by SharasGit** [comment]
Tested on Paper-83 and it's not working. Same behavior.

**Comment by nicegamer7** [comment]
I couldn't reproduce this with the latest build of Spigot from BuildTools. Can you try with Spigot?

**Comment by jinkhya** [comment]
It doesn't work on latest build of Spigot for me nor Paper#83.
```

>version
[17:39:36] [Server thread/INFO]: This server is running CraftBukkit version git-Spigot-c3a49df-dba91d2 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)
```

I tried making a fresh end with multiverse with spigot running instead of paper but the same happens. After defeating that dragon the gateway doesn't teleport you to endcities range.

**Comment by jinkhya** [comment]
I also tried this : 
- reset bukkit.yml and spigot.yml
- disabled all plugins but Multiverse-core
- reset configs of Multiververse-core completely
- made a complete fresh new world
and still can't get it to work ; what's your secret :)

**Comment by R416alex** [comment]
Confirmed for me on a test server with just mv plugins.

**Comment by nicegamer7** [comment]
Sorry, let's just clarify.

Do the gateways do nothing? or teleport you to the wrong spot?

**Comment by R416alex** [comment]
> Sorry, let's just clarify.
> 
> Do the gateways do nothing? or teleport you to the wrong spot?

They don't do anything for me. I can be standing in the gateway block and it doesn't do anything. No error messages either.

**Comment by NeverGrowUp** [comment]
Confirmed for me too. Since Paper#83, the return portal from End to Overworld is fixed, but the End Gateways do nothing at all. Even using the trapdoor method (rather than enderpearl), and climbing inside the gateway portal, nothing happens. No errors/stacks in console.

**Comment by SoRadGaming** [comment]
Me too, only the default world (the one in server.properties) works. Interestingly the outer end gateways do teleport you back to the main island and work as intended, it is simply the end gateway that generates after killing the ender dragon that does not work.

**Comment by roopekoo** [comment]
I am having this issue also. End gateways from the main island do not work. But end gateways outside to main island work as they should.

**Comment by csev1755** [comment]
I'm having the same problem. Everything works fine in the original end world file created from the first run. When I use mvcreate to make an end world, I can not go through any of the gateways.

My exact process (tried in Spigot and Paper)
1. Execute server jar file for first time to create base world
2. Stop server, drop Multiverse jars into plugin folder
3. Run server again, use /mv create test end
4. Join and use /mvtp test
5. Beat dragon and attempt to use gateway

**Comment by nicegamer7** [comment]
Could someone please test this again with the latest Spigot build or with Paper Build 93, the report back whether it's still happening and the output of `/version`?

**Comment by NerdyKyogre** [comment]
> Could someone please test this again with the latest Spigot build or with Paper Build 93, then report back on whether it's still happening and with the output of `/version`?

Don't know exactly what build I'm on but I updated a couple hours ago and this is still a thing.

**Comment by roopekoo** [comment]
I was using paper-93 and end gateways are still broken.
/version
`[19:57:08 INFO]: This server is running Paper version git-Paper-93 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)`
`[19:57:08 INFO]: Previous version: git-Paper-83 (MC: 1.16.1)`
`[19:57:08 INFO]: You are running the latest version`
[Me chilling in the end gateway](https://imgur.com/a/VbAPdz2)

**Comment by roopekoo** [comment]
Also tried on latest spigot:
/version
`[20:20:04] [Server thread/INFO]: This server is running CraftBukkit version git-Spigot-05da6fa-75a2fb8 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)`
`[20:20:04] [Server thread/INFO]: Checking version, please wait...`
`[20:20:06] [Thread-12/INFO]: You are running the latest version`

Also tried this with 
Multiverse-Core-4.1.1-SNAPSHOT build 777 with no luck.
In conclusion, end gateways are broken in 1.16.1 in multiverse in spigot and paper.

**Comment by NickatGIHS** [comment]
Just to clarify:
If i have a server created in 1.15.2 (Spigot) where an end world _has been visited,_ if I were to update to 1.16.1 everything would work?

**Comment by nicegamer7** [comment]
I was able to reproduce the bug, and have filed it here, https://hub.spigotmc.org/jira/browse/SPIGOT-6011.

**Comment by nicegamer7** [comment]
It's been marked as fixed. Please update Spigot using BuildTools or use Paper Build 110 (or higher) and see if it's working for you.

**Comment by csev1755** [comment]
It's working for me on Paper #110. Thanks!

**Comment by NeverGrowUp** [comment]
Can double confirm. Paper #110 fixes end gateways for me too. Thanks nicegamer7!

**Comment by nicegamer7** [comment]
Awesome! I'll assume it's working in Spigot then.
